lib/dma.c

```diff
@@ -445,24 +477,16 @@ int dma_controller_dirty_page_logging_start(dma_controller_t *dma, size_t pgsize
 
     for (i = 0; i < (size_t)dma->nregions; i++) {
         dma_memory_region_t *region = &dma->regions[i];
-        ssize_t bitmap_size;
-
-        bitmap_size = get_bitmap_size(region->info.iova.iov_len, pgsize);
-
-        if (bitmap_size < 0) {
-            return bitmap_size;
-        }
-        region->dirty_bitmap = calloc(bitmap_size, sizeof(char));
-        if (region->dirty_bitmap == NULL) {
-            int ret = errno;
+        if (dirty_page_logging_start_on_region(region, pgsize) < 0) {
```

Comment:
don't we want this to be idempotent?

Reply:
Do you mean effective ignoring calls to `dma_controller_dirty_page_logging_start` if dirty page tracking has already started? If so, then this is checked right before this `for` loop.